jobs/payment-jobs/tasks/ejv_partner_distribution_task.py

```diff
@@ -53,308 +54,221 @@ def create_ejv_file(cls):
+                                                 ejv_header_model: EjvHeaderModel,
+                                                 sequence: int):
+        """Update disbursement status and create EJV Link."""
+        if target.is_legacy:
+            invoice = InvoiceModel.find_by_id(target.identifier)
+            invoice.disbursement_status_code = DisbursementStatus.UPLOADED.value
+        else:
+            # Only EFT is using partner disbursements table for now, eventually we want to move our disbursement
+            # process over to something similar: Where we have an entire table setup that
+            # is used to track disbursements, instead of just the three column approach that
+            # doesn't work when there are multiple reversals etc.
+            partner_disbursement = PartnerDisbursementsModel.find_by_id(target.identifier)
+            partner_disbursement.status_code = DisbursementStatus.UPLOADED.value
+            partner_disbursement.processed_on = datetime.now(tz=timezone.utc)
+
+        db.session.add(EjvLinkModel(link_id=target.identifier,
+                                    link_type=target.disbursement_type,
+                                    ejv_header_id=ejv_header_model.id,
+                                    disbursement_status_code=DisbursementStatus.UPLOADED.value,
+                                    sequence=sequence))
 
     @classmethod
     def _get_partners_by_batch_type(cls, batch_type) -> List[CorpTypeModel]:
         """Return partners by batch type."""
         # CREDIT : Ministry GL code -> disbursement_distribution_code_id on distribution_codes table
         # DEBIT : BC Registry GL Code -> distribution_code on fee_schedule, starts with 112
         bc_reg_client_code = current_app.config.get('CGI_BCREG_CLIENT_CODE')  # 112
+        # Rule for GA. Credit is 112 and debit is 112.
+        # Rule for GI. Debit is 112 and credit is not 112.
         query = db.session.query(DistributionCodeModel.distribution_code_id) \
             .filter(DistributionCodeModel.stop_ejv.is_(False) | DistributionCodeModel.stop_ejv.is_(None)) \
             .filter(DistributionCodeModel.account_id.is_(None)) \
-            .filter(DistributionCodeModel.disbursement_distribution_code_id.is_(None))
-
-        if batch_type == 'GA':
-            # Rule for GA. Credit is 112 and debit is 112.
-            partner_distribution_code_ids: List[int] = db.session.scalars(query.filter(
-                DistributionCodeModel.client == bc_reg_client_code
-            )).all()
-        else:
-            # Rule for GI. Debit is 112 and credit is not 112.
-            partner_distribution_code_ids: List[int] = db.session.scalars(query.filter(
-                DistributionCodeModel.client != bc_reg_client_code
-            )).all()
+            .filter(DistributionCodeModel.disbursement_distribution_code_id.is_(None)) \
+            .filter_conditionally(True if batch_type == 'GA' else None,
```

Comment:
work around for `filter_conditionally`

might just make another `filter_boolean`